src/test/ui/const-eval/ub-enum-ptr.stderr

```diff
@@ -0,0 +1,10 @@
+error: this constant cannot be used
+  --> $DIR/ub-enum-ptr.rs:23:1
+   |
+LL | const BAD_ENUM: Enum = unsafe { Foo { a: &1 }.b}; //~ ERROR this constant cannot be used
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ a raw memory access tried to access part of a pointer value as raw bytes
```

Comment:
We can't report out of bounds before we know the bits, reading the bits will result in this error though ;)

Reply:
Hm, that error is not great though because nothing here tries to read the pointer as bytes, Also, I don't think I understand which line in `validate_scalar` triggers this -- the `Scalar::Ptr(_)` case doesn't do any reads.

IOW, how is this failure at all related to `Enum` having `0` as its only value?

Reply:
I found the issue...

```rust
enum Foo {
    Bar = 0,
}
```

has `Tagged` variants, so reading the tag will bug out.

I'll do what you proposed all along and move the `Abi` check up (just for `Uninhabited` and `Scalar`).

Reply:
Ah so it's a `Scalar` but also `Tagged`? Makes sense.

We'd still get that error though of it's an `enum` with data. Could we get a more specific error message like "invalid: enum discriminant is a pointer" or "error determining enum discriminant" in that case?

Reply:
sure, I can just run the scalar-checks on the discriminant before trying to read the discriminant.

Reply:
Hm, I thought of a simple `.map_err`, but I have no idea what would work better.